### PR TITLE
Update multiformats, add JWS utilities, update deps

### PR DESCRIPTION
**Pull request by rvagg**
breaking because `encodeIdentityCID()` is async now

Will follow up with a PR to jd-dag-jose that couples with (and uses) this and discuss over there.

Custom Jest resolver is pending fix being tracked @ https://github.com/facebook/jest/issues/9771

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,54 +13,76 @@ export interface EncodedPayload {
...
+export async function prepareCleartext(
+  cleartext: Record<string, any>,
+  blockSize?: number
+): Promise<Uint8Array> {
+  return pad((await encodeIdentityCID(cleartext)).bytes, blockSize)
 }
 
 export function decodeCleartext(b: Uint8Array): Record<string, any> {
-  return decodeIdentityCID(unpadCIDBytes(b))
+  return decodeIdentityCID(CID.decodeFirst(b)[0])
```
Comment (oed) [inline src/index.ts:87]: Nice, this seems like a new useful thing :)

File: README.md [hunk 2]
```diff
@@ -46,10 +46,14 @@ Encode an ipld object as a CID that uses the identity hash.
 ### decodeIdentityCID(cid: CID): Record<string, any>
 Decode an ipld object from a CID that uses the identity hash.
 
+### toJWSPayload(payload: EncodedPayload | CID): string
+Transform an `EnvodedPayload` (from `encodePayload()`) or a CID into a JWS string for use with `createJWS()` in [did-jwt](https://github.com/decentralized-identity/did-jwt). The string form is simply the Base64url encoded form of the CID's byte representation.
+
+### toJWSStrings(jose: any): string[]
```
Comment (oed) [inline README.md:52]: Unfortunate that you need this method. Ideally `did-jwt` should support the General encoding, that the ´jose` object here has, out of the box!

File: README.md [hunk 3]
```diff
@@ -46,10 +46,14 @@ Encode an ipld object as a CID that uses the identity hash.
...
+Transform a `DagJWT` object from ipld-dag-jose into an array of strings for each signature in the object. The strings can then be verified using `verifyJWS()` in [did-jwt](https://github.com/decentralized-identity/did-jwt).
```
Comment (oed) [inline README.md]:
```suggestion
Transform a `DagJWS` object from ipld-dag-jose into an array of strings for each signature in the object. The strings can then be verified using `verifyJWS()` in [did-jwt](https://github.com/decentralized-identity/did-jwt).
```

File: src/index.ts [hunk 4]
```diff
@@ -14,54 +13,76 @@ export interface EncodedPayload {
...
 
+export function toJWSPayload(payload: EncodedPayload | CID): string {
+  let cid = CID.asCID(payload)
+  if (!cid) {
+    cid = CID.asCID(payload.cid)
+  }
+  if (!cid) {
+    throw new Error('Payload must be an EncodedPayload or a CID')
+  }
+  return base64url.encode(cid.bytes).slice(1)
```
Comment (oed) [inline src/index.ts:31]: Does `base64url.encode` return multibase and that's why we slice?

File: README.md [hunk 5]
```diff
@@ -46,10 +46,14 @@ Encode an ipld object as a CID that uses the identity hash.
 ### decodeIdentityCID(cid: CID): Record<string, any>
 Decode an ipld object from a CID that uses the identity hash.
 
+### toJWSPayload(payload: EncodedPayload | CID): string
+Transform an `EnvodedPayload` (from `encodePayload()`) or a CID into a JWS string for use with `createJWS()` in [did-jwt](https://github.com/decentralized-identity/did-jwt). The string form is simply the Base64url encoded form of the CID's byte representation.
```
Comment (oed) [inline README.md]:
```suggestion
Transform an `EncodedPayload` (from `encodePayload()`) or a CID into a JWS string for use with `createJWS()` in [did-jwt](https://github.com/decentralized-identity/did-jwt). The string form is simply the Base64url encoded form of the CID's byte representation.
```

File: src/index.ts [hunk 6]
```diff
@@ -14,54 +13,76 @@ export interface EncodedPayload {
...
+}
+
+// map a DagJWT to an array of JWT strings that verifyJWT() will be able to verify
```
Comment (oed) [inline src/index.ts]:
```suggestion
// map a DagJWS to an array of JWS strings that verifyJWS() will be able to verify
```

**Review by oed — COMMENTED** [review]
Very nice, been looking forward to using the new multiformats libs.

Can you bump the package version to 1.0?

Should we wait with merging this until the jest issue has been resolved?

File: src/index.ts (hunk 4 above)
Comment (rvagg) [reply src/index.ts:31]: yeah, and there's no easy way to get to the plain base64 encoder (yet—this has annoyed me on a few occasions so I think I might go and do something about it)

**Comment by rvagg** [comment]
version bumped, typos fixed

> Should we wait with merging this until the jest issue has been resolved?

No, I went down the rabbit hole on that one and the dependency chain of things that need to be resolved is deep and unlikely to be resolved any time soon unfortunately. A large collection of users are having to work around some of the Jest problems so maybe pressure will increase to find a more speedy resolution but for now it's on hold.

**Review by oed — APPROVED** [review]
lgtm!

**Comment by oed** [comment]
`dag-jose-utils@1.0.0` published 🎉

**Pull request merged**